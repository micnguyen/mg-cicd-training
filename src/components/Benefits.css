.benefits-container {
  background-color: rgba(255, 255, 255, 0.1);
  border-radius: 12px;
  padding: 2rem;
  max-width: 600px;
  box-shadow: 0 4px 6px rgba(0, 0, 0, 0.1);
}

.benefits-intro {
  font-size: 1.2rem;
  margin-bottom: 1.5rem;
  line-height: 1.6;
}

.benefits-list {
  text-align: left;
  padding-left: 1rem;
}

.benefit-item {
  margin-bottom: 1rem;
  font-size: 1.1rem;
  line-height: 1.5;
}
