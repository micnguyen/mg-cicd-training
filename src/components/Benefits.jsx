import React from 'react';
import './Benefits.css';

function Benefits() {
  const benefits = [
    "Fasterererer delivery of features to production",
    "Automated testing reduces human error",
    "Reduces bugs",
    "Immediate feedback on code quality",
    "Repeatable & consistent deployment process",
    "Just awesome software",
  ];

  return (
    <div className="benefits-container" data-testid="benefits-container">
      <p className="benefits-intro">
        Continuous Integration and Continuous Deployment streamlines software development by:
      </p>
      <ul className="benefits-list">
        {benefits.map((benefit, index) => (
          <li key={index} className="benefit-item">{benefit}</li>
        ))}
      </ul>
    </div>
  );
}

export default Benefits;
