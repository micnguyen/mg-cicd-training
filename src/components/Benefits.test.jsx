// src/components/Benefits.test.jsx
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';  // Add this import
import Benefits from './Benefits';

describe('Benefits Component', () => {
  test('renders the benefits container', () => {
    render(<Benefits />);
    const benefitsContainer = screen.getByTestId('benefits-container');
    expect(benefitsContainer).toBeInTheDocument();
  });

  test('renders the benefits intro text', () => {
    render(<Benefits />);
    const introText = screen.getByText(/Continuous Integration and Continuous Deployment/i);
    expect(introText).toBeInTheDocument();
  });

  test('renders all benefit items', () => {
    render(<Benefits />);
    expect(screen.getByText(/Faster delivery of features/)).toBeInTheDocument();
    expect(screen.getByText(/Automated testing/)).toBeInTheDocument();
    expect(screen.getByText(/Immediate feedback/)).toBeInTheDocument();
    expect(screen.getByText(/Repeatable & consistent deployment process/)).toBeInTheDocument();
    expect(screen.getByText(/Just awesome22 software/)).toBeInTheDocument();
  });
});