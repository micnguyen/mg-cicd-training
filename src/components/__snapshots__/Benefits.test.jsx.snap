// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Benefits Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="benefits-container"
    data-testid="benefits-container"
  >
    <p
      class="benefits-intro"
    >
      Continuous Integration and Continuous Deployment streamlines software development by:
    </p>
    <ul
      class="benefits-list"
    >
      <li
        class="benefit-item"
      >
        Faster delivery of features to production
      </li>
      <li
        class="benefit-item"
      >
        Automated testing reduces human error
      </li>
      <li
        class="benefit-item"
      >
        Immediate feedback on code quality
      </li>
      <li
        class="benefit-item"
      >
        Repeatable & consistent deployment process
      </li>
      <li
        class="benefit-item"
      >
        Just awesome software
      </li>
      <li
        class="benefit-item"
      >
        Reduces bugs
      </li>
      <li
        class="benefit-item"
      >
        Another Entry woops some extra text
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
